app/frontend/src/components/designer/FormViewerMultiUpload.vue

```diff
@@ -0,0 +1,535 @@
+      }
+    },
+    handleFile() {
+      if (this.file == undefined) {
+        this.$refs.file.$refs.input.click();
+        // this.$refs.file.click();
+      }
+    },
+    removeFile(file) {
+      this.files = this.files.filter((f) => {
+        return f != file;
+      });
+    },
+    parseFile() {
+      try {
+        let reader = new FileReader();
+        reader.onload = function (e) {
+          this.Json = JSON.parse(e.target.result);
+        }.bind(this);
+
+        reader.onloadend = () => {
+          this.preValidateSubmission();
+        };
+        reader.readAsText(this.file);
+      } catch (e) {
+        this.resetUpload();
+        this.addNotification({
+          message: this.ERROR.PARSE_JSON_ERROR,
+          consoleError: e,
+        });
+      }
+    },
+    preValidateSubmission() {
+      try {
+        if (!Array.isArray(this.Json)) {
+          this.resetUpload();
+          this.addNotification({
+            message: this.ERROR.JSON_OBJECT_NOT_ARRAY,
+            consoleError: 'An unexpected error occurred.',
+          });
+          return;
+        }
+        if (this.Json.length == 0) {
+          this.resetUpload();
+          this.addNotification({
+            message: this.ERROR.JSON_ARRAY_EMPTY,
+            consoleError: 'this file is empty.',
+          });
+          return;
+        }
+        this.globalError = [];
+        this.index = 0;
+        this.max = 100;
+        this.progress = true;
+        this.$emit('toggleBlock', true);
+        this.validate(this.Json[this.index], []);
+      } catch (error) {
+        this.resetUpload();
+        this.$emit('set-error', {
+          error: true,
+          message: this.ERROR.ERROR_WHILE_VALIDATE,
+        });
+        this.addNotification({
+          message: this.ERROR.ERROR_WHILE_VALIDATE,
+          consoleError: error,
+        });
+        return;
+      }
+    },
+    validate(element, errors) {
+      const timer = setTimeout(
+        function () {
+          try {
+            let newForm = this.formElement;
+            newForm.data = element;
+            newForm.submission.data = element;
+            newForm.triggerChange();
+            let validationResult = newForm.checkValidity();
+            if (!validationResult) {
+              errors.push({
+                submission: this.index,
+                errors: validationResult.errors,
+              });
+            }
+          } catch (error) {
+            errors.push({
+              submission: this.index,
+              message: this.ERROR.ERROR_WHILE_CHECKVALIDITY,
+            });
+          }
+          this.index++;
+          this.value = this.pourcentage(this.index);
+          clearTimeout(timer);
+          if (this.index < this.Json.length) {
+            this.validate(this.Json[this.index], errors);
+          } else {
+            this.endValidation(errors);
+          }
+        }.bind(this),
+        12
+      );
+    },
+    pourcentage(i) {
```

Comment:
please try to keep the language consistent across the application. this should be `percentage`.

Reply:
Solved